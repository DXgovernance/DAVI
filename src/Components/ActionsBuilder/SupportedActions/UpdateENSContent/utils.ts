import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
