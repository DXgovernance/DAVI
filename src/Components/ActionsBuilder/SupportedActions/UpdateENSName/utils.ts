import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import { useState, useEffect } from 'react';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIPFSHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const useDebounce = (value: string, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);

  useEffect(() => {
    const handler = setTimeout(() => {
      setDebouncedValue(value);
    }, delay);

    return () => {
      clearTimeout(handler);
    };
  }, [value, delay]);

  return debouncedValue;
};
