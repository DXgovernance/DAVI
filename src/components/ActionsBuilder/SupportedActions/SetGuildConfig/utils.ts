import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
