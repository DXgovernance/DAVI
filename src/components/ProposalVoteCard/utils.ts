import { ConfirmVoteProposalProps, VoteOnProposalProps } from './types';

export const voteOnProposal = ({
  votingPowerAtProposalSnapshot,
  votingPowerAtProposalCurrentSnapshot,
}: VoteOnProposalProps) => {
  const hasNoVotingPower =
    votingPowerAtProposalSnapshot &&
    Number(votingPowerAtProposalSnapshot?.toString()) <= 0;

  const hasVotingPowerAtCurrentSnapshot =
    votingPowerAtProposalCurrentSnapshot &&
    Number(votingPowerAtProposalCurrentSnapshot?.toString()) > 0;
  return {
    hasNoVotingPower,
    hasVotingPowerAtCurrentSnapshot,
  };
};

export const confirmVoteProposal = ({
  proposal,
  contract,
  selectedOption,
  userVotingPower,
  createTransaction,
  cb,
}: ConfirmVoteProposalProps) => {
  createTransaction(
    `Vote on proposal ${proposal?.title}`,
    async () => contract.setVote(proposal?.id, selectedOption, userVotingPower),
    true,
    cb
  );
};

export const getOptionLabel = ({ metadata, optionKey, t }) => {
  const metadataLabel = metadata?.voteOptions?.[optionKey];
  return metadataLabel
    ? metadataLabel
    : Number(optionKey) === 0
    ? t('against', { defaultValue: 'Against' })
    : t('option', { optionKey });
};
