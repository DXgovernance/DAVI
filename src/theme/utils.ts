import { css } from 'styled-components';

export const SUPPORTED_THEMES = {
  DARK: 'DARK',
  LIGHT: 'LIGHT',
};

const MEDIA_WIDTHS = {
  upToSmall: 600,
  upToMedium: 960,
  upToLarge: 1280,
};

export const mediaWidthTemplates = Object.keys(MEDIA_WIDTHS).reduce(
  (accumulator, size) => {
    accumulator[size] = (...args: any[]) => css`
      @media (max-width: ${MEDIA_WIDTHS[size]}px) {
        ${
          // @ts-ignore
          css(...args)
        }
      }
    `;
    return accumulator;
  },
  {}
);
