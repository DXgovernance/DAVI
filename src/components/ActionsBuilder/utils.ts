import { DecodedAction } from './types';
import { Permission } from './types';
import Web3 from 'web3';
import { NULL_SIGNATURE } from 'utils';
import { SupportedAction } from './types';

const web3 = new Web3();

export const getPermissionArgs = (
  decodedActions: DecodedAction[]
): Permission[] => {
  const permissions = decodedActions.map(decodedAction => {
    // Native transfer of ETH does not have a function
    if (
      decodedAction?.decodedCall?.callType ===
        SupportedAction.NATIVE_TRANSFER ||
      decodedAction?.decodedCall?.callType === SupportedAction.RAW_TRANSACTION
    ) {
      return {
        from: decodedAction?.decodedCall?.from,
        to: decodedAction?.decodedCall?.to,
        callType: decodedAction?.decodedCall?.callType,
        functionSignature: NULL_SIGNATURE,
      };
    }
    const functionInputs = decodedAction?.decodedCall?.function?.inputs?.map(
      input => input.type
    );
    const functionName = decodedAction?.decodedCall?.function?.name;
    const functionSignature = web3.eth.abi.encodeFunctionSignature(
      `${functionName}(${functionInputs.join(',')})`
    );
    return {
      from: decodedAction?.decodedCall?.from,
      to: decodedAction?.decodedCall?.to,
      callType: decodedAction?.decodedCall?.callType,
      functionSignature,
    };
  });
  return permissions;
};
