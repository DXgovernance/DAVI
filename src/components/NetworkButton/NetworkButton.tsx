import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { getChainIcon } from 'utils';
import { Button, ButtonIcon, IconButton } from 'components/primitives/Button';
import { NetworkModal } from 'components/Web3Modals/NetworkModal';
import { useAccount, useNetwork } from 'wagmi';

const NetworkButton = () => {
  const { t } = useTranslation();
  const [isNetworkModalOpen, setIsNetworkModalOpen] = useState(false);
  const { isConnected } = useAccount();
  const { chain } = useNetwork();

  const toggleNetworkModal = () => {
    setIsNetworkModalOpen(!isNetworkModalOpen);
  };

  function getNetworkStatus() {
    if (isConnected && chain?.unsupported) {
      return (
        <Button onClick={toggleNetworkModal}>{t('unsupportedNetwork')}</Button>
      );
    } else if (isConnected) {
      return (
        <IconButton
          data-testid="change-network-btn"
          onClick={toggleNetworkModal}
          iconLeft
        >
          <ButtonIcon src={getChainIcon(chain.id)} alt={'Icon'} />
          {chain.name}
        </IconButton>
      );
    } else {
      return <Button onClick={toggleNetworkModal}>{t('notConnected')}</Button>;
    }
  }

  return (
    <>
      {getNetworkStatus()}
      <NetworkModal isOpen={isNetworkModalOpen} onClose={toggleNetworkModal} />
    </>
  );
};

export default NetworkButton;
