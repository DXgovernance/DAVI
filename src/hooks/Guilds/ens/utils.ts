// import { isSupportedChainId } from 'components/ActionsBuilder/SupportedActions/UpdateENSContent/utils';
import { MAINNET_ID } from 'utils';

export const isAvailableOnENS = (chainId: number) => {
  const ensNetworks = [MAINNET_ID];
  return ensNetworks.includes(chainId);
};
