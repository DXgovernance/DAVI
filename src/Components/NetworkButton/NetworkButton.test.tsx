import NetworkButton from './NetworkButton';
import { render } from 'utils/tests';

describe.skip('NetworkButton', () => {
  it('Should match snapshot and display not connected status', () => {
    const { container, getByText } = render(<NetworkButton />);
    expect(container).toMatchSnapshot();
    expect(getByText('notConnected')).toBeInTheDocument();
  });
  it('Should match snapshot and display network', () => {
    const { container, getByText } = render(<NetworkButton />);
    expect(container).toMatchSnapshot();
    expect(getByText('Ethereum Mainnet')).toBeInTheDocument();
  });
});
