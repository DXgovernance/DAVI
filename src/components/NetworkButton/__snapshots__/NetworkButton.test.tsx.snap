// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkButton Should match snapshot and display network 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  font-size: 14px;
  border: 1px solid #303338;
  background-color: #295FF4;
  color: #fff;
  border-radius: 32px;
  padding: 0.5rem 0.8rem;
  margin: 0.2rem;
  width: auto;
}

.c0:disabled {
  color: initial;
  opacity: 0.4;
  cursor: auto;
}

.c0:hover:enabled {
  border-color: #fff;
}

.c0:active:enabled {
  border: 1px solid #303338;
}

.c1 svg,
.c1 img {
  margin-right: .4rem;
}

.c1,
.c1 button {
  padding: .25rem 0.6rem;
  background-color: #1B1D1F;
}

.c2 {
  height: 1.6rem;
  width: 1.6rem;
}

<div>
   
  <button
    class="c0 c1"
    data-testid="changeNetworkBtn"
  >
    <img
      alt="Icon"
      class="c2"
    />
    Random Chain
  </button>
   
</div>
`;

exports[`NetworkButton Should match snapshot and display not connected status 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  font-size: 14px;
  border: 1px solid #303338;
  background-color: #295FF4;
  color: #fff;
  border-radius: 32px;
  padding: 0.5rem 0.8rem;
  margin: 0.2rem;
  width: auto;
}

.c0:disabled {
  color: initial;
  opacity: 0.4;
  cursor: auto;
}

.c0:hover:enabled {
  border-color: #fff;
}

.c0:active:enabled {
  border: 1px solid #303338;
}

<div>
   
  <button
    class="c0"
  >
    notConnected
  </button>
   
</div>
`;
